src/editor/JSTTextView.m

```diff
@@ -15,6 +15,10 @@
 
 static NSString *JSTQuotedStringAttributeName = @"JSTQuotedString";
 
+@protocol NSAppearanceChocolate
+@property (nonatomic, readonly) BOOL isDark_bc;
```

Comment:
I'm not super happy with this, don't know what's the best solution

Reply:
I think it would be better to have `JSTTextView` have a theme property, which takes values from an enum:

```
typedef NS_ENUM(NSUInteger, JSTTextViewTheme) {
  JSTTextViewThemeLight = 0,
  JSTTextViewThemeDark,
  JSTTextViewThemeDefault = JSTTextViewThemeLight,
};
```

I would set the editor's theme where we make the text view (somewhere in Sketch, I _think_). That view would also need to switch the theme if the user changes the OS theme whilst it's open (`viewDidChangeEffectiveAppearance`?).

Reply:
That's a good idea but we use the view in an NSWindowController which doesn't have `viewDidChangeEffectiveAppearance`. Any idea what would be the alternative?

Reply:
You could use KVO to observe the window's `effectiveAppearance`.

```
static void * windowAppearanceContext = &windowAppearanceContext;

- (void)windowDidLoad {
    [self.window addObserver:self forKeyPath:@"effectiveAppearance" options:(NSKeyValueObservingOptionNew) context:windowAppearanceContext];
…
}

- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context {
  if (context == windowAppearanceContext) {
    // set the editor theme based on self.window.effectiveAppearance
  }
}

```

…the only problem with that is when to unregister for the notification. I think it's best to make the `NSWindowController` the window's delegate (if it isn't already), and do that in `windowWillClose:`

```
- (void)windowWillClose:(NSNotification *)notification {
  [self.window removeObserver:self forKeyPath:@"effectiveAppearance" context:windowAppearanceContext];
}
```